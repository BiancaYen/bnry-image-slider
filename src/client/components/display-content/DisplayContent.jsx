// Display Content

import PropTypes from 'prop-types';

// Types
import { ElseType } from '../else/Else';

// Default Props
const defaultProps = {
    isDisplayed: false
};

// Prop Types
const propTypes = {
    children: PropTypes.node.isRequired,
    isDisplayed: PropTypes.bool
};

const DisplayContent = ({ isDisplayed, children }) => {
    if (isDisplayed) {
        return children.length ? children.filter(child => child.type.componentType !== ElseType) : children;
    }
    return children.length ? children.filter(child => child.type.componentType === ElseType) : null;
};

DisplayContent.defaultProps = defaultProps;
DisplayContent.propTypes = propTypes;

export default DisplayContent;
