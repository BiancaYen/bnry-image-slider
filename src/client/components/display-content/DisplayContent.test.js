// DisplayContent Test

import { mount, shallow } from 'enzyme';
import React from 'react';

// Components
import DisplayContent from './DisplayContent';

// Constants
const content = 'Content';

describe('DisplayContent', () => {
    it('renders with all required prop types', () => {
        const component = shallow(
            <DisplayContent isDisplayed>
                <div>
                    {content}
                </div>
            </DisplayContent>
        );

        expect(component).toMatchSnapshot();
    });

    it('asserts that content renders correctly depending on prop type', () => {
        const component = mount(
            <DisplayContent isDisplayed>
                <div>
                    {content}
                </div>
            </DisplayContent>
        );

        expect(component.find('div').text()).toEqual(content);
    });

    it('asserts that content renders correctly depending on prop type', () => {
        const component = mount(
            <DisplayContent isDisplayed={false}>
                <div>
                    {content}
                </div>
            </DisplayContent>
        );

        expect(component.html()).toEqual(null);
    });
});
