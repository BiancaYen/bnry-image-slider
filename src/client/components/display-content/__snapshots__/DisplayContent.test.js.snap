// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayContent renders with all required prop types 1`] = `
<div>
  Content
</div>
`;
